UefiPayloadPkg/UefiPayloadPkg.dsc

```diff
@@ -436,6 +443,21 @@
   gEfiMdeModulePkgTokenSpaceGuid.PcdStatusCodeUseMemory|FALSE
   gEfiMdeModulePkgTokenSpaceGuid.PcdUse1GPageTable|TRUE
 
+  #
+  # Build the PcdDebugPropertyMask from build flags
+  #

```

Comment:
python UefiPayloadPkg/UniversalPayloadBuild.py 
--pcd gEfiMdeModulePkgTokenSpaceGuid.PcdStatusCodeUseSerial==TRUE
--pcd gEfiMdePkgTokenSpaceGuid.PcdDebugPropertyMask=0x2F

Do we really need to add the macro ?

Reply:
I think of it more as something specific to in-memory debug logs. In my MinPlatform port, I use similar for a library I wrote. What do you think of:
```
DEFINE IN_MEMORY_DEBUG_LOGS = FALSE
DEFINE USE_CBMEM_FOR_CONSOLE = FALSE

!if $(USE_CBMEM_FOR_CONSOLE) == TRUE
DEFINE IN_MEMORY_DEBUG_LOGS = TRUE
!endif

!if $(USE_CBMEM_FOR_CONSOLE) == TRUE
# define library
!endif

!if $(IN_MEMORY_DEBUG_LOGS) == TRUE
# define PCD overrides
!endif
```

If you don't want CbSerialPortLib in UefiPayloadPkg, an idea I had is moving it to a module (in PlatformPayloadFeaturePkg, wherever) and calling StatusCodeHandler protocol to register it. But I'd want to wait until UPL is fully adopted.

I want to support IN_MEMORY_DEBUG_LOGS as a configuration here. Even if coreboot should pass both IN_MEMORY_DEBUG_LOGS and USE_CBMEM_FOR_CONSOLE, debug logs are often important even when we're not using debug systems. On my personal system, I use gEfiMdePkgTokenSpaceGuid.PcdDebugPropertyMask=0x03 or gEfiMdePkgTokenSpaceGuid.PcdDebugPropertyMask=0x07. In coreboot, the platform builder can choose debug logs even without selecting a bigger debug build, because it's a helpful start to debug anything.

Reply:
I think 

before UefiUPL (Legacy Uefi Payload)
- You can use build --pcd XXXX=<value> to customized PCD value to meet your requirement.

after UefiUPL (Uefi Universal Payload)
- You can use UniversalPayloadBuild.py --pcd XXXX=<value> to customized PCD value to meet your requirement.

For long term, fully adopt to Uefi UPL, Coreboot specific change, need to move to Edk2Platform.

Because based on this change you just want to change below PCDs settings value.
- gEfiMdePkgTokenSpaceGuid.PcdDebugPropertyMask

And use RELEASE_LOGGING to control CC_FLAGS I don't have concern.